include/z64camera.h

```diff
@@ -27,6 +27,73 @@
 #define PARENT_CAM(cam) ((cam)->globalCtx->cameraPtrs[(cam)->parentCamId])
 #define CHILD_CAM(cam) ((cam)->globalCtx->cameraPtrs[(cam)->childCamId])
 
+#define CAM_IFACE_ALPHA(alpha) ((alpha) << 8)
+
+/**
+ * useShrinkWindowNow: boolean - if true, set shrinkbox immediately. if false, set target. see CAM_SHRINKWIN_CURVAL
+ * shinkWindowFlag: determines the size of the letter-box shrink window. See CAM_SHRINKWINVAL_* enums
+ * interfaceAlpha: hides certain hud icons
+ *    - A value of 0 in camera is translated to an interface alpha of 50, which is the value to restore all hud icons to the screen
+ *    - A value of 0xF in camera results in no change in the alpha
+ * funcFlags: Custom flags for functions
+ */
+#define CAM_INTERFACE_FLAGS(useShrinkWindowNow, shinkWindowFlag, interfaceAlpha, funcFlags) \
```

Comment:
I may be overcomplicating this... it's quite a mess

Thoughts on passing the 0xF000 part to `CAM_INTERFACE_FLAGS` and building this part with another macro if needed (`CAM_SHRINKWIN_FLAG` below):

```c
#define CAM_INTERFACE_FLAGS(shrinkWindowFlag, interfaceAlpha, funcFlags) \
    (((shrinkWindowFlag) & CAM_SHRINKWIN_MASK) | CAM_IFACE_ALPHA((interfaceAlpha) & 0xF) | ((funcFlags) & 0xFF))

enum {
    SHRINKWINVAL_NONE,
    SHRINKWINVAL_SMALL,
    SHRINKWINVAL_MEDIUM,
    SHRINKWINVAL_LARGE
};

#define CAM_SHRINKWIN_FLAG(instantaneous, size) (((size) << 12) | ((instantaneous) ? CAM_SHRINKWIN_CURVAL : 0))
```

```c
// usage
CAM_INTERFACE_FLAGS(CAM_SHRINKWIN_FLAG(true, SHRINKWINVAL_SMALL), ...)
CAM_INTERFACE_FLAGS(CAM_SHRINKWINVAL_IGNORE, ...)
```

idk this looks too complicated...

```c
#define CAM_INTERFACE_FLAGS(shrinkWindowFlag, interfaceAlpha, funcFlags) \
    (((shrinkWindowFlag) & CAM_SHRINKWIN_MASK) | CAM_IFACE_ALPHA((interfaceAlpha) & 0xF) | ((funcFlags) & 0xFF))

CAM_INTERFACE_FLAGS(CAM_SHRINKWINVAL_NONE, 0, 0);
CAM_INTERFACE_FLAGS(CAM_SHRINKWINVAL_NONE | CAM_SHRINKWIN_CURVAL, 0, 0);
```

looks more like a better suggestion, especially since `CAM_SHRINKWIN_CURVAL` is nearly never used

Reply:
So on the user-side going from:
`CAM_INTERFACE_FLAGS(false, CAM_SHRINKWINVAL_NONE, 5, NORM1_FLAG_2 | NORM1_FLAG_1)`
to
`CAM_INTERFACE_FLAGS(CAM_SHRINKWIN_FLAG(false, CAM_SHRINKWINVAL_NONE), 5, NORM1_FLAG_2 | NORM1_FLAG_1)`
which does group the boolean to shinkwinval without looking up what it is, but at the cost of making it even more verbose, I'm not sure I prefer it. It does get a bit weird though how giving the value `0xF000` basically ignores the `instantaneous` bool so there is a bit of a grouping there, but I don't think `CAM_SHRINKWIN_FLAG(` helps clear things up.

Also, I guess it's nicer to define `SHRINKWINVAL_` as enums then shift it later, but then it doesn't change anything from the user side and you'd have to add extra shifts where the function actually reads those flags

I think using `CAM_SHRINKWIN_CURVAL` instead of `<< 31` is good though

Reply:
Also keep in mind the size of this macro might grow later if we can get some kind of macro going for interfaceAlpha's (I've been brainstorming those and have a rough idea but it's ugly so we'll see if it going anywhere or not)